rules/js.txt

```diff
@@ -2270,6 +2285,45 @@
+
+<new DOMString> = <FontFace>.src;
+<FontFace>.src = <FontSrc>;
+<new FontFace> = new FontFace(<DOMString>, <FontSrc>, {});
+<new FontFace> = new FontFace(<DOMString>, <FontSrc>, {<FontFaceDescriptor>});
+<new FontFace> = new FontFace(<DOMString>, <FontSrc>, {<FontFaceDescriptor>, <FontFaceDescriptor>, <FontFaceDescriptor>});
+
+#FontFaceDescriptor
+<new FontFaceDescriptor> = "ascent-override: normal";
```

Comment:
Note that you can put these in jshelpers.txt (somewhere before `!begin helperlines`) without the `new` attribute, which would then inline the descriptor without creating a new string variable for it.

Also, since the values here actually come from font-face CSS rule (see https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face), a more comprehensive patch would be to extend the CSS grammar to support the `@font-face` rule (it does not, currently) and then here include relevant values from the CSS grammar (possible because `jshelper.txt` includes `cssproperties.txt`. This way, these would also be emitted in the generated CSS. Or we can leave it as is for now, but add a comment to extend CSS grammar in the future.

Reply:
Made this change: added `@font-face` rule to css and moved all the descriptors into `cssproperties.txt`